File: packages/reliai-python/reliai/replay.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any

from .client import ReliaiClient, get_default_client


@dataclass(slots=True)
class ReliaiReplayPipeline:
    payload: dict[str, Any]

    def run(self) -> dict[str, Any]:
        return self.payload


def replay(trace_id: str, client: ReliaiClient | None = None) -> ReliaiReplayPipeline:
    active_client = client or get_default_client()
    payload = active_client.request_json(f"/api/v1/traces/{trace_id}/replay")
    return ReliaiReplayPipeline(payload=payload)

File: packages/reliai-node/src/replay.ts
import { getDefaultClient, type ReliaiClient } from "./client";

export interface ReliaiReplayStep {
  span_id: string;
  parent_span_id: string | null;
  span_name: string | null;
  span_type: string;
  inputs: Record<string, unknown> | null;
  template: string | null;
  variables: Record<string, unknown> | null;
  model: string | null;
  parameters: Record<string, unknown> | null;
  prompt: string | null;
  tool_name: string | null;
  guardrail_policy: string | null;
  guardrail_action: string | null;
}

export interface ReliaiReplayPayload {
  trace_id: string;
  project_id: string;
  environment: string;
  steps: ReliaiReplayStep[];
}

export class ReliaiReplayPipeline {
  constructor(readonly payload: ReliaiReplayPayload) {}

  run(): ReliaiReplayPayload {
    return this.payload;
  }
}

export async function replay(traceId: string, client: ReliaiClient = getDefaultClient()): Promise<ReliaiReplayPipeline> {
  const payload = await client.requestJson<ReliaiReplayPayload>(`/api/v1/traces/${traceId}/replay`);
  return new ReliaiReplayPipeline(payload);
}
